### ISSUE TITLE
Variables with same name as javascript keyword causes unexpected behavior.

### ISSUE THREAD

**Issue by thorjelly** [elm-lang/elm-compiler#1304]
For example, the variable name 'default' used in the following function causes the entire compiled application to be able to run, because it is a javascript keyword and causes unexpected behavior in the compiled script:

```
getBest : List ( Float ) -> (Float -> Float -> Bool) -> Float -> Float
getBest list predicate default =
    case list of
        (x::xs) -> getBest xs predicate (if predicate x default then x else default)
        [] -> default
```
In this case, it seems to prevent the elm module from loading. Firefox reports "elmModule is undefined" and chrome reports "Cannot read property 'make' of undefined", breaking the entire application because of the name of the variable.

Not sure if other cases exist, but perhaps other situations can arise with other variables named the same as js keywords. Either way, it seems like it should be considered a bug in elm, since it causes no elm compile error but the resulting application does not work and gives an unexplained javascript error.

**Comment by jvoigtlaender** [comment]
Thanks for opening the issue. I only now realized that your function is tail recursive (and that `default` is already contained in the compiler's `jsReserveds` set). So it's possible I have already fixed your case, with pull request https://github.com/elm-lang/elm-compiler/pull/1251, which has meanwhile been merged, but not yet released with a new compiler version.

**Comment by avh4** [comment]
Thanks for the report -- probably fixed by the PR Janis mentioned.  Should be verified before 0.17 release.

**Comment by Momus** [comment]
Just ran across this doing Exercism.io: http://exercism.io/submissions/08fe0ad53e7b4021ac4815f981e55695
using the word "function."
Since I made this same mistake on another exercise, I noticed that when running elm-test, I always got the same JS error, which I'm reproducing here for Googleability:
```js
    accumulate $ elm-test AccumulateTests.elm 
Success! Compiled 1 modules.
Successfully generated /tmp/elm_test_116311-2502-1yl1rbp.js
Successfully compiled AccumulateTests.elm
Running tests...
[stdin]:9070
               function = _v1;
                        ^

SyntaxError: Unexpected token =
    at Object.exports.runInThisContext (vm.js:53:16)
    at Object.<anonymous> ([stdin]-wrapper:6:22)
    at Module._compile (module.js:409:26)
    at node.js:579:27
    at nextTickCallbackWith0Args (node.js:420:9)
    at process._tickCallback (node.js:349:13)
```
Will look forward to testing  the fix when 0.17 comes out.

**Comment by jvoigtlaender** [comment]
It's indeed fixed. Just verified that, for both @thorjelly and @Momus's examples. So this issue can be closed.
